src/app/core/components/members-dialog/members-dialog.component.ts

```diff
@@ -134,6 +134,23 @@ export class MembersDialogComponent implements OnInit, IsTabbedDialog {
   }
 
   async onAddMemberClick() {
+    const archive = this.accountService.getArchive();
+    if (
+      archive.accessRole !== 'access.role.manager' && archive.accessRole !== 'access.role.owner'
+    ) {
+      this.promptService.confirm(
+        'Learn More',
+        'Add Member',
+        null,
+        null,
+        `You do not have permission to add members to this archive.`
+      ).then(() => {
```

Comment:
This method is already marked `async`, and we have other `await`ed code. Can we structure this as a `try`-`catch` block instead of as promises?